test/framework/test_composite.py

```diff
@@ -1000,3 +1017,127 @@ def circuits(self):
         self.assertExperimentDone(meta_expdata)
         job_ids = meta_expdata.job_ids
         self.assertEqual(len(job_ids), 2)
+
+
+class TestNewWorkflow(QiskitExperimentsTestCase):
+
+    """
+    #1268
+    """
+
+    def setUp(self):
+
+        super().setUp()
+
+        # NOTE: When I use setUp for calculating variables in
+        # exp_data disappear. I dont know why.
+
+        backend = FakeBackend()
+
+        exp1 = FakeExperiment([0])
+        exp2 = FakeExperiment([1])
+        par_exp1 = ParallelExperiment([exp1, exp2], flatten_results=True)
+
+        exp3 = FakeExperiment([0])
+        exp4 = FakeExperiment([1])
+        par_exp2 = ParallelExperiment([exp3, exp4], flatten_results=True)
+
+        # Set a batch experiment
+        batch_exp = BatchExperiment([par_exp1, par_exp2], flatten_results=True)
+        self.exp_data = batch_exp.run(backend)
+
+    def test_new_workflow_is_done(self):
```

Comment:
I don't think this test is necessary. Instead you can add another test case of running exp_data.run two times and check if the result is the same. The `FakeAnalysis` attached to `FakeExperiment` doesn't perform analysis depending on the circuit data, it probably better to write custom analysis for this test and check if circuit data is properly passed (and processed) to the component analyses.